### ISSUE TITLE
Need help with understanding the issue: "Invalid or unsupported XPath"

### ISSUE THREAD

**Issue by stanislaw** [seleniumbase/SeleniumBase#2192]
First, thank you for creating and maintaining `selenium_base`. I have been using it very successfully for one year now!

I started getting this issue yesterday only on the GitHub Actions Python 3.7-based macOS-based job. The Linux and Windows Python 3.7-based jobs work fine. My macOS- and Python 3.7-based job works.

Full stack trace:

```
============================== 1 failed in 0.28s ===============================
FAILED                      [100%]
strictdoc/core/test_file_tree.py:3 (test_xpath)
def test_xpath():
        # Works
        xpath = '//sdoc[@data-testid="node-requirement"][3]//*[@data-testid="node-edit-action"]'
        convert_xpath_to_css(xpath)
    
        # Does not work
        xpath = '//sdoc-node[@data-testid="node-requirement"][3]//*[@data-testid="node-edit-action"]'
>       convert_xpath_to_css(xpath)

strictdoc/core/test_file_tree.py:11: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/Stanislaw/.pyenv/versions/3.7.15/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/seleniumbase/fixtures/xpath_to_css.py:228: in convert_xpath_to_css
    css = _get_raw_css_from_xpath(xpath, original)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

xpath = '//sdoc-node[@data-testid="node-requirement"][3]/descORself/*[@data-testid="node-edit-action"]'
original = '//sdoc-node[@data-testid="node-requirement"][3]//*[@data-testid="node-edit-action"]'

    def _get_raw_css_from_xpath(xpath, original):
        css = ""
        attr = ""
        position = 0
    
        while position < len(xpath):
            node = prog.match(xpath[position:])
            if node is None:
                raise XpathException(
                    "\nInvalid or unsupported XPath:\n%s\n"
                    "(Unable to convert XPath Selector to CSS Selector)"
>                   "" % original
                )
E               seleniumbase.fixtures.xpath_to_css.XpathException: 
E               Invalid or unsupported XPath:
E               //sdoc-node[@data-testid="node-requirement"][3]//*[@data-testid="node-edit-action"]
E               (Unable to convert XPath Selector to CSS Selector)

/Users/Stanislaw/.pyenv/versions/3.7.15/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/seleniumbase/fixtures/xpath_to_css.py:92: XpathException
```

I tried to reduce it to a minimal case and it looks like the function does not like the `-` symbol. Here is my failing CI job just in case, where the two tests fail because of this issue: https://github.com/strictdoc-project/strictdoc/actions/runs/6519580587/job/17706990342?pr=1332. There, the code reaches this function from `hover_and_click`.

```
from seleniumbase.fixtures.xpath_to_css import convert_xpath_to_css


def test_xpath():
    # Works
    xpath = '//sdoc[@data-testid="node-requirement"][3]//*[@data-testid="node-edit-action"]'
    convert_xpath_to_css(xpath)

    # Does not work
    xpath = '//sdoc-node[@data-testid="node-requirement"][3]//*[@data-testid="node-edit-action"]'
    convert_xpath_to_css(xpath)
```

Given that the other jobs are still working fine, I suspect that this code is only reached on macOS under a certain condition, but I couldn't investigate that just yet. In any case, it seems reasonable that the `convert_xpath_to_css` function would support the tag names with `-` in them.

Please let me know if I can provide more information.

Thank you!

**Comment by mdmintz** [comment]
Thanks for reporting this. I created a bug ticket for it: https://github.com/seleniumbase/SeleniumBase/issues/2193
(I'll have a fix released soon.)

**Comment by mdmintz** [comment]
This was resolved in `4.20.3` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.20.3
